### ISSUE TITLE
Update weights on all edged connected to a node

### ISSUE THREAD

**Issue by fredrik-jansson-se** [petgraph/petgraph#614]
### Summary

I try to, given a node, find all outgoing edges, and update the weight on those edges.

I can't find any docs on how to do this and can't figure out the code. This doesn't compile:

```
   let output = self.graph[node_idx].calculate_output(&[], self.pos, self.dir, world);
   let mut nbrs = self
        .graph
        .edges_directed(node_idx, petgraph::Direction::Outgoing);
   for n in nbrs { 
        self.graph[n.id()] = output; 
   }
```

```
error[E0502]: cannot borrow `self.graph` as mutable because it is also borrowed as immutable
   --> src/organizm.rs:182:17
    |
178 |               let mut nbrs = self
    |  ____________________________-
179 | |                 .graph
    | |______________________- immutable borrow occurs here
180 |                   .edges_directed(node_idx, petgraph::Direction::Outgoing);
181 |               for n in nbrs {
    |                        ---- immutable borrow later used here
182 |                   self.graph[n.id()] = output;
    |                   ^^^^^^^^^^ mutable borrow occurs here
```

There doesn't seem to be any `.detached()` function on the `edges_directed` iterator.

### Details

* New docs or maybe an `edges_directed_mut` function that returns an iterator where I can change the edge weights.

**Comment by JuraxH** [comment]
@fredrik-jansson-se I believe that in your case, instead of `edges_iterator` iterator you should use [`WalkNeighbors`](https://docs.rs/petgraph/latest/petgraph/graph/struct.WalkNeighbors.html) that does not borrow the graph, docs for this structure have an example similar to what you are doing.

To compile your code I would replace the `edges_directed` with `neighbors_directed` and used `detach` on it to get the walker:
```
   let output = self.graph[node_idx].calculate_output(&[], self.pos, self.dir, world);
   let mut nbrs = self
        .graph
        .neighbors_directed(node_idx, petgraph::Direction::Outgoing)
        .detach();
   while let Some(edge) = nbrs.next_edge(&g) { 
        self.graph[edge] = output; 
   }
```

**Comment by fredrik-jansson-se** [comment]
Got it, thank you for the help!

With that, I'll close the issue.
